cloudinit/config/cc_set_passwords.py

```diff
@@ -79,14 +79,33 @@
 PW_SET = "".join([x for x in ascii_letters + digits if x not in "loLOI01"])
 
 
-def handle_ssh_pwauth(pw_auth, distro):
+def handle_ssh_pwauth(pw_auth, distro: Distro):
     """Apply sshd PasswordAuthentication changes.
 
     @param pw_auth: config setting from 'pw_auth'.
                     Best given as True, False, or "unchanged".
     @param distro: an instance of the distro class for the target distribution
 
     @return: None"""
+    service = distro.get_option("ssh_svcname", "ssh")
+    try:
+        distro.manage_service("status", service)
+    except subp.ProcessExecutionError as e:
+        LOG.warning(
+            (
+                "Ignoring config 'ssh_pwauth: %s'. SSH deamon service '%s' is "
```

Comment:
> Ideally I think we want a pre-flight check at the beginning of handle_ssh_pwauth function if the ssh_svcname in question does not exist for restarting. If not present, exit early with a LOG.warning(f"Ignoring config 'ssh_pwauth: {pw_auth}'. Service {service} is not installed")

I though that it was more generic to say that the service is not running, as a non 0 exit from `systemctl status service` no always does mean that the service is not installed.